### PR TITLE
feat: adds field validation and onChange events

### PR DESCRIPTION
**Pull request by kevinperaza**
## Description

- feat: adds element validation
- feat: adds onChange events
- refactor: reconcile masking

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change
-

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import 'react-native-url-polyfill/auto';
```
Comment (kevinperaza) [inline src/index.ts:1]: We need this; need to add it back

**Review by lcschy — APPROVED** [review]

**Comment by kevinperaza** [comment]
Dependencies/Code checks are failing due to https://github.com/axios/axios/issues/6006

**Review by lcschy — APPROVED** [review]

**Pull request merged**